src/Symfony/Validator/State/ParameterValidatorProvider.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Symfony\Validator\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use ApiPlatform\Validator\Exception\ValidationException;
+use Symfony\Component\Validator\ConstraintViolation;
+use Symfony\Component\Validator\ConstraintViolationList;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * Validates parameters using the symfony validator.
+ *
+ * @experimental
+ */
+final class ParameterValidatorProvider implements ProviderInterface
+{
+    public function __construct(
+        private readonly ProviderInterface $decorated,
+        private readonly ValidatorInterface $validator
+    ) {
+    }
+
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
+    {
+        $body = $this->decorated->provide($operation, $uriVariables, $context);
+        if (!$context['request'] ?? null) {
+            return $body;
+        }
+
+        $operation = $context['request']->attributes->get('_api_operation');
+        foreach ($operation->getParameters() as $parameter) {
+            if ($parameter->getRequired() && !\array_key_exists('_api_values', $parameter->getExtraProperties())) {
+                throw new ValidationException(new ConstraintViolationList([new ConstraintViolation(sprintf('Parameter "%s" is required.', $parameter->getKey()), null, [], null, null, null)]));
+            }
```

Comment:
Can't this constraint be automatically set in `ParameterResourceMetadataCollectionFactory` instead?

Reply:
Indeed, I'll try a default value to `null` with a NotBlank constraint?

Reply:
Maybe just add NotBlank(allowNull: false) if `required` option is set?